docs/lang/articles/advanced/layout.md

```diff
@@ -100,54 +126,50 @@ int y[2][3];  // column-major
 
 for (int i = 0; i < 3; i++) {
     for (int j = 0; j < 2; j++) {
-        do_something( x[i][j] );
-        do_something( y[j][i] );
+        do_something(x[i][j]);
+        do_something(y[j][i]);
     }
 }
 ```
 
 :::
 
-## Array of Structures (AoS), Structure of Arrays (SoA)
+### Array of Structures (AoS) versus Structure of Arrays (SoA)
 
-Fields of same size can be placed together.
+Fields of same shape can be placed together.
 
-For example, this places two 1D fields of size `3` (array of structure, AoS):
+For example, the following places two 1D fields of shape `3` together, which
+is called Array of Structures (AoS):
 
 ```python
 ti.root.dense(ti.i, 3).place(x, y)
 ```
 
-Their memory layout:
+Their memory layout is:
 
 ```
-#  address low ............. address high
-#  x[0]   y[0] | x[1]  y[1] | x[2]   y[2]
+#  address: low ......................... high
+#           x[0]  y[0]  x[1]  y[1]  x[2]  y[2]
 ```
 
-By contrast, this places two field placed separately (structure of array, SoA):
+By contrast, the following places these two fields separately, which is called
+Structure of Arrays (SoA):
 
 ```python
 ti.root.dense(ti.i, 3).place(x)
 ti.root.dense(ti.i, 3).place(y)
 ```
 
-Now, their memory layout:
+Now their memory layout is:
 
 ```
-#  address low ............. address high
-#  x[0]  x[1]   x[2] | y[0]   y[1]   y[2]
+#  address: low ......................... high
+#           x[0]  x[1]  x[2]  y[0]  y[1]  y[2]
 ```
 
-Normally, you don't have to worry about the performance nuances between
-different layouts, and should just define the simplest layout as a
-start. However, locality sometimes have a significant impact on the
-performance, especially when the field is huge.
-
-**To improve spatial locality of memory accesses (i.e. cache hit rate /
-cacheline utilization), it's sometimes helpful to place the data
-elements within relatively close storage locations if they are often
-accessed together.** Take a simple 1D wave equation solver for example:
+**To improve spatial locality of memory accesses, it's sometimes helpful to
+place the data elements within relatively close addresses if they are often
+accessed together.** Take a simple 1D wave equation solver as an example:
```

Comment:
```suggestion
**To improve spatial locality of memory accesses, it may be helpful to
place data elements that are often accessed together within
relatively close addresses.** Take a simple 1D wave equation solver as an example:
```